### DISCUSSION TITLE
MQTT runs into an exception published by AMQP 0-9-1 publishers with an expiration property set

### DISCUSSION THREAD

**Discussion by rdsilio** [rabbitmq/rabbitmq-server#12707] [Other] [answered]
### Describe the bug

Messages with expiration property cause mqtt plugin to crash

### Reproduction steps

Subscribe to an mqtt queue
Post an AMQP message with the expiration property to an mqtt queue

### Expected behavior

The message should be consumed normally

![image](https://github.com/user-attachments/assets/75faa902-d350-493d-abd9-71050c82b871)

![image](https://github.com/user-attachments/assets/a7e81907-d3fc-4f59-ad8c-36a945b81ceb)

**Comment by rdsilio** [comment]
Problem found in versions 3.13.x to 4.0.3

**Comment by michaelklishin** [comment]
@rdsilio please use Discussions for questions and investigations.

We need an executable way to reproduce, that is, publishing should come from a client library, not a browser/HTTP API client.

**Comment by michaelklishin** [comment]
According to @mkuratczyk, with an upcoming version of [`omq`](https://github.com/rabbitmq/omq), this scenario should be reproducible with

```
omq amqp-mqtt --message-ttl 10s -t /topic/my_topic -T my_topic -C 1
```

**Reply by mkuratczyk** [reply]
Available now, in 0.16

**Comment by michaelklishin** [comment, marked as answer]
https://github.com/rabbitmq/rabbitmq-server/pull/12710
https://github.com/rabbitmq/rabbitmq-server/pull/12708

**Comment by keycad** [comment]
Hi, I see this has been solved for RabbitMQ 4.0.x and 4.1.x, but it hasn't been solved for 3.13.x, which has commercial support until dec 2029. We face this issue and we would really appreciate this hotfix being included in 3.13.x. I know community support for this version ended, but is there any chance this fix could be included in 3.13.x? Thanks

**Reply by ansd** [reply]
This is fixed in VMware Tanzu RabbitMQ `v3.13.x`, but not in open-source RabbitMQ `v3.13.x`.

@keycad please create an issue using official Broadcom support channels and we will make sure that this gets backported to open-source RabbitMQ `v3.13.x` as well.

**Reply by ansd** [reply]
If you do not hold a commercial license, you can contact sales via the `CONTACT SALES` button in https://www.vmware.com/products/app-platform/tanzu-data-intelligence/rabbitmq

**Reply by ansd** [reply]
I backported the fix to open source RabbitMQ. The fix will be available in 3.13.16.

**Reply by keycad** [reply]
Thank you very much @ansd !
